### PR TITLE
Features/629+826 unittest mozbuild updates

### LINKED ISSUE #629: Unit Testing -Help needed!
The code base is growing and we have zero unit testing. Retrofitting this is not easy, so going forward we should be unit testing as we commit.
Any tests will be accepted, unit or functional. There is no code coverage requirement, just hit the major use cases.

### PR DESCRIPTION
**Pull request by crankycoder**
This fixes #629 to add JUnit, robolectric and jococo coverage analysis
#826 needed a unittest around the Updater code, but that's actually going away because of the play store, so this actually closes off #862 as well.

I need to add some documentation for this, but the short of it is that the Makefile has reasonable targets.

The default `all` target will build a release.
`test` will compile and run unittests

There is now a split between debug and unittest builds as robolectric puts extra constraints on what is possible.

### CODE REVIEW

File: android-library/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest
+	package="cz.posvic.bootstrap.lib"
```
Comment (crankycoder) [inline android-library/src/main/AndroidManifest.xml:3]: Rename this to org.mozilla.mozstumbler

File: android-library/src/main/res/menu/main.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+	xmlns:app="http://schemas.android.com/apk/res-auto"
+	xmlns:tools="http://schemas.android.com/tools"
+	tools:context="cz.posvic.bootstrap.lib.LibraryActivity">
```
Comment (crankycoder) [inline android-library/src/main/res/menu/main.xml:4]: rename this to org.mozilla.mozstumbler as well.  just grep for the string and replace all instances.

**Comment by crankycoder** [comment]
@garvankeeley I'm just waiting for travis to get back before merging this.

**Comment by garvankeeley** [comment]
On a fresh pull I am getting modifications locally:
On branch dev
Your branch is up-to-date with 'origin/dev'.

```
deleted:    MozStumbler.iml
modified:   android-library/android-library.iml
modified:   android/android.iml
```

**Comment by garvankeeley** [comment]
Will this commit also break the updater system? 
android/src/main/java/org/mozilla/mozstumbler/client/Updater.java 29:  
private static final String VERSION_URL = "https://raw.github.com/mozilla/MozStumbler/master/VERSION";

**Comment by garvankeeley** [comment]
And the app icon is gone.

**Comment by crankycoder** [comment]
Let's change the VERSION_URL to point to something that _isn't_ in source control.  If we accidentally push a broken change into source control, I don't want to deal with hundreds of broken clients.

Let's have it point to some github wiki page instead.  I've opened ticket #865 to review and rework the Updater code.

**Comment by crankycoder** [comment]
I still have the real app icon, maybe a stale build?  I'll double check.  Those pngs I dropped look like the stock generic android robot icons.

**Comment by crankycoder** [comment]
@garvankeeley The icon seems to be ok for me - I've deleted all my build directories and reconstructed the APK and I'm still getting the new-ish stumbler icon.  The icon is referenced from the AndroidManifest.xml file here:

https://github.com/mozilla/MozStumbler/blob/dev/android/src/main/AndroidManifest.xml#L29

Which should just pull the proper ic_launcher.png file from the resources directory.

**Comment by crankycoder** [comment]
VERSION file re-added in #871

**Comment by garvankeeley** [comment]
I got the icon to work once I copied it to the XDPI resources.

Stripping the icon from android-library project would be good to avoid confusion.

**Pull request merged**